src/python/pants/init/engine_initializer.py

```diff
@@ -369,7 +369,7 @@ def setup_legacy_graph_extended(
 
     @rule
     def glob_match_error_behavior_singleton() -> GlobMatchErrorBehavior:
-      return glob_match_error_behavior or GlobMatchErrorBehavior.ignore
```

Comment:
We soon plan to deprecate `GlobMatchErrorBehavior.ignore`. This `or` was only being triggered in tests, so it's safe to change to default to `warn` without an API change.